lib/liquid/spec/yaml_initializer.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/string/output_safety"
+
+class SerializableProc
+  class << self
+    def new(og_code, args: [])
+      code = "proc do |" + args.join(",") + "|\n" + og_code + "\nend"
+      p = eval(code) # rubocop:disable Security/Eval
+      p.instance_variable_set(:@og_code, og_code)
+
+      p
+    end
+  end
+end
+
+class Proc
+  def encode_with(coder)
+    coder.represent_scalar("!serializable_proc", @og_code)
+  end
+
+  def _dump(level)
+    @og_code
+  end
+  class << self
+    def _load(args)
+      SerializableProc.new(args)
+    end
+  end
+end
+
+# Custom deserialization for Proc objects
+YAML.add_domain_type("", "serializable_proc") do |_, val|
+  SerializableProc.new(val)
+end
+
+YAML.add_domain_type("", "stub_exception_renderer") do |_, val|
+  StubExceptionRenderer.new(raise_internal_errors: val["raise_internal_errors"])
+end
+
+class ShopifyFileSystem
```

Comment:
I guess it's fine to prefix "Shopify", but ideally naming would be neutral (more instances below).

Reply:
I think this is fine as it mimics a bit closer what you would expect a shopify theme to do